scipy/linalg/_decomp_update.pyx

```diff
@@ -0,0 +1,2416 @@
+        blas_pointers.dgemm(transa, transb, &m, &n, &k, &alpha, a, &lda,
+                b, &ldb, &beta, c, &ldc)
+    elif blas_t is float_complex:
+        blas_pointers.cgemm(transa, transb, &m, &n, &k, &alpha, a, &lda,
+                b, &ldb, &beta, c, &ldc)
+    else:
+        blas_pointers.zgemm(transa, transb, &m, &n, &k, &alpha, a, &lda,
+                b, &ldb, &beta, c, &ldc)
+
+cdef inline void trmm(char* side, char* uplo, char* transa, char* diag, int m,
+        int n, blas_t alpha, blas_t* a, int lda, blas_t* b, int ldb) nogil:
+    if blas_t is float:
+        blas_pointers.strmm(side, uplo, transa, diag, &m, &n, &alpha, a, &lda,
+                b, &ldb)
+    elif blas_t is double:
+        blas_pointers.dtrmm(side, uplo, transa, diag, &m, &n, &alpha, a, &lda,
+                b, &ldb)
+    elif blas_t is float_complex:
+        blas_pointers.ctrmm(side, uplo, transa, diag, &m, &n, &alpha, a, &lda,
+                b, &ldb)
+    else:
+        blas_pointers.ztrmm(side, uplo, transa, diag, &m, &n, &alpha, a, &lda,
+                b, &ldb)
+
+cdef inline int geqrf(int m, int n, blas_t* a, int lda, blas_t* tau,
+                      blas_t* work, int lwork) nogil:
+    cdef int info
+    if blas_t is float:
+        lapack_pointers.sgeqrf(&m, &n, a, &lda, tau, work, &lwork, &info)
+    elif blas_t is double:
+        lapack_pointers.dgeqrf(&m, &n, a, &lda, tau, work, &lwork, &info)
+    elif blas_t is float_complex:
+        lapack_pointers.cgeqrf(&m, &n, a, &lda, tau, work, &lwork, &info)
+    else:
+        lapack_pointers.zgeqrf(&m, &n, a, &lda, tau, work, &lwork, &info)
+    return info
+
+cdef inline int ormqr(char* side, char* trans, int m, int n, int k, blas_t* a,
+    int lda, blas_t* tau, blas_t* c, int ldc, blas_t* work, int lwork) nogil:
+    cdef int info = 0 
+    if blas_t is float:
+        lapack_pointers.sormqr(side, trans, &m, &n, &k, a, &lda, tau, c, &ldc,
+                work, &lwork, &info)
+    elif blas_t is double:
+        lapack_pointers.dormqr(side, trans, &m, &n, &k, a, &lda, tau, c, &ldc,
+                work, &lwork, &info)
+    elif blas_t is float_complex:
+        lapack_pointers.cunmqr(side, trans, &m, &n, &k, a, &lda, tau, c, &ldc,
+                work, &lwork, &info)
+    else:
+        lapack_pointers.zunmqr(side, trans, &m, &n, &k, a, &lda, tau, c, &ldc,
+                work, &lwork, &info)
+    return info
+
+#------------------------------------------------------------------------------
+# Utility routines
+#------------------------------------------------------------------------------
+
+cdef cnp.ndarray PyArray_FromArraySafe(cnp.ndarray arr, void* newtype, int flags):
+    """In Numpy 1.5.1, Use of the NPY_F_CONTIGUOUS flag is broken when used
+    with a 1D input array.  The work around is to pass NPY_C_CONTIGUOUS since
+    for 1D arrays these are equivalent.  This is Numpy's gh-2287, fixed in
+    9b8ff38.
+
+    FIXME: Is it worth only applying this for numpy 1.5.1? 
```

Comment:
That commit is in numpy 1.6.2, which is now the lowest supported version. So this can go.